### PR TITLE
Add paths for homebrew on Apple silicon

### PR DESCRIPTION
**Pull request by eyeseast**
This also passes in the extension path when specified in GIS methods. Wherever we know an extension path, we use `db.init_spatialite(find_spatialite() or load_extension)`.

----
:books: Documentation preview :books:: https://sqlite-utils--531.org.readthedocs.build/en/531/

### CODE REVIEW

File: sqlite_utils/cli.py [hunk 1]
```diff
@@ -1407,7 +1407,7 @@ def create_database(path, enable_wal, init_spatialite, load_extension):
 
     # load spatialite from expected locations and initialize metadata
     if init_spatialite:
-        db.init_spatialite()
+        db.init_spatialite(find_spatialite() or load_extension)
```
Comment (simonw) [inline sqlite_utils/cli.py:1410]: I don't understand this pattern.

The `init_spatialite()` function already calls `find_spatialite()` if necessary:

https://github.com/simonw/sqlite-utils/blob/fc221f9b62ed8624b1d2098e564f525c84497969/sqlite_utils/db.py#L1161-L1165

The `load_extension` option is a bit different: It's actually a list of extensions passed using `--load-extension path` on the command-line, which is used by this code:

https://github.com/simonw/sqlite-utils/blob/fc221f9b62ed8624b1d2098e564f525c84497969/sqlite_utils/cli.py#L3037-L3047

**Comment by simonw** [comment]
Aah, I think I see why you wrote it like that.

The problem is that `init_spatialite()` does other stuff too:

https://github.com/simonw/sqlite-utils/blob/fc221f9b62ed8624b1d2098e564f525c84497969/sqlite_utils/db.py#L1161-L1171

So it needs to be able to load the SpatiaLite extension from the correct place, and THEN run `select InitSpatialMetadata()` to configure the database, if needed.

So the problem you're trying to solve here is to let people optionally pass in the path to SpatiaLite if it's not one of the ones that are searched by default.

**Comment by eyeseast** [comment]
Exactly, that's what I was running into. On my M2 MacBook, SpatiaLite ends up in what is -- for the moment -- a non-standard location, so even when I passed in the location with `--load-extension`, I still hit an error on `create-spatial-index`.

What I learned doing this originally is that SQLite needs to load the extension for each connection, even if all the SpatiaLite stuff is already in the database. So that's why `init_spatialite()` gets called again.

Here's the code where I hit the error: https://github.com/eyeseast/boston-parcels/blob/main/Makefile#L30 It works using this branch.

I'm not attached to this solution if you can think of something better. And I'm not sure, TBH, my test would actually catch what I'm after here.

**Comment by eyeseast** [comment]
I'm going to close this in favor of #536. Will try a cleaner approach to custom paths once that one is merge.

**Pull request closed without merging**